### ISSUE TITLE
Policy/guideline modification request for long options

### ISSUE THREAD

**Issue by joehays** [tldr-pages/tldr#12125]
### Command description

I would like the community to reconsider a TLDR guideline/policy documented in [here](https://github.com/tldr-pages/tldr/blob/main/CONTRIBUTING.md)

Specifically, one of the policies/guidelines states,

`Try to incorporate the spelled-out version of single-letter options in the example's description. The goal is to allow people to understand the syntax of the commands, not just memorize it.`

I'd argue that this needs to be tweaked. Yes, understanding is the priority but I argue that most users will not take the time to write out the long spelled-out version simply because it takes too long. So, the current policy is not teaching everything that the user wants to learn. It's failing to achieve the an equally important goal... teaching the reader how to use a command quickly. 

TLDR used to give the short-hand versions but it changed... now the tool is less valuable to me because I now have to open `man` to look up what I really want to learn (the faster short-hand version). 

So... I suggest that we change the policy/guideline to document both the spelled-out version and shorter/faster version one right after another (spelled out first followed by the short-hand version). Then both learning objectives will be satisfied. 

For example,  one entry in the tldr for grep could be,

```
 - Search for a pattern in all files recursively in a directory, showing line numbers of matches, ignoring binary files:
    grep --recursive --line-number --binary-files=without-match "search_pattern" path/to/directory
    grep -RnI "search_pattern" path/to/directory
```
This is a modest increase in text but makes TLDR more valuable. 

### Command details

it's a suggested policy/guideline change 

### Platform

Common

**Comment by o0nd7ots** [comment]
> This is a modest increase in text but makes TLDR more valuable.

And if we want the volume increase to be smaller, we can introduce short options one at a time, either with explicit conversion `-{short|-long}` or just `-short`


# git archive

> Create an archive of files from a tree.
> More information: <https://git-scm.com/docs/git-archive>.

- Create a tar archive from the contents of the current HEAD and print it to `stdout`:

`git archive --verbose HEAD`

- Use the `zip` format and report progress [v]erbosely:

`git archive -{{v|-verbose}} --format zip HEAD`

- [o]utput the zip archive to a specific file:

`git archive -v -{{o|-output}} {{path/to/file.zip}} HEAD`

- Create a `tar` archive from the contents of the latest commit of a specific branch:

`git archive -o {{path/to/file.tar}} {{branch_name}}`

- Use the contents of a specific directory:

`git archive -o {{path/to/file.tar}} HEAD:{{path/to/directory}}`

- Prepend a path to each file to archive it inside a specific directory:

`git archive -o {{path/to/file.tar}} --prefix {{path/to/prepend}}/ HEAD`

**Comment by sbrl** [comment]
> I'd argue that this needs to be tweaked. Yes, understanding is the priority but I argue that most users will not take the time to write out the long spelled-out version simply because it takes too long.

This has always bugged me since longer options were preferred. The original intention was never to prefer long options to the exclusion of short options. It's always a case-by-case basis, but in many cases there's a compelling argument for short options, e.g. `grep` which is a commonly used command.

As you yourself mention, often what you actually what is to learn the short option if you're regularly using the terminal.

**Mnemonics:** The original intention here was to integrate the `[s]yntax` into the description to aid the memory when **short options were used** (e.g. as in `tar`), not as a substitute for including short options in the description instead of the command!

Further, the suggestion in #12059 to use `-{{o|-output}}` as an alternative format does seem reasonable to me also if there's a need to display both, and it shouldn't break clients.

The [style guide](https://github.com/tldr-pages/tldr/blob/main/contributing-guides/style-guide.md#option-syntax) really needs updating in this respect. To this end, I propose the following guidelines:

1. For frequently used commands (e.g. `grep`, `tar`, etc), prefer short options + mnemonic.
2. For everything else, prefer `{{-o|-output}}` syntax.
3. Commands one can reasonably expect most terminal users to rarely execute, prefer long options as before.

Thoughts?

**Comment by joehays** [comment]
@sbrl  you've got my vote!

**Comment by sbrl** [comment]
> Regarding guideline 2, I think you mean `-{{o|-output}}`? Otherwise it would read as `-o` OR `-output`.

Yep! Thanks for catching that one - comment updated @sebastiaanspeck.
